src/gui/torrentcontentmodel.cpp

```diff
@@ -113,31 +106,22 @@ namespace
         {
             const QString ext = info.suffix();
             if (!ext.isEmpty()) {
-                const QPixmap pixmap = pixmapForExtension(ext, QSize(32, 32));
-                if (!pixmap.isNull())
-                    return QIcon(pixmap);
+                auto cacheEntry = m_iconCache.find(ext);
+                
+                if (cacheEntry != m_iconCache.end())
+                    return cacheEntry.value();
+                
+                QIcon icon = QIcon(pixmapForExtension(ext, QSize(32, 32)));
+                if (!icon.isNull())
```

Comment:
Braces is correct. The coding style I'm used to has braces being used on every if statement even when they're one line... mainly so I don't have to worry about making silly mistakes like I just did.